### ISSUE TITLE
Save vectors in remote HDFS

### ISSUE THREAD

**Issue by paulomann** [spotify/annoy#445]
Suppose I have two machines: **M1**, and **M2**, **M1** has the HDFS file system, while **M2** is where my Machine Learning inference code is running. What I need to do is to store vectors in the HDFS file system (in **M1**), and retrieve all similar vectors in **M2**. Since the number of vectors can rapidly increase, the **M2** will not have sufficient space, but the HDFS file system will have.

TL;DR: Can I store Annoy vectors in a remote HDFS and read them in another machine?

**Comment by erikbern** [comment]
Annoy only supports local file system, but you might be able to mount it using some driver. I don’t know if mmap is supported for these custom file systems though

**Comment by piskvorky** [comment]
IIRC mmap is possible for HDFS, but fulfilling page faults from a remote machine is a bad idea anyway (terrible performance).

**Comment by erikbern** [comment]
That’s exactly right

**Comment by paulomann** [comment]
Yes @piskvorky, I agree. Thank you both, @erikbern.
